content/tutorials/drift-detection/index.md

```diff
@@ -0,0 +1,198 @@
+
+1. Add your GitHub organization/repository and branch you created.
+
+2. Add your Secret environment variables, including `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`.
+
+{{% notes type="info" %}}
+Secret environment variables, such as `AWS_SECRET_ACCESS_KEY`, are encrypted end-to-end with Pulumi and can be set on each stack. However, by creating an [environment](/docs/pulumi-cloud/esc) with Pulumi ESC, you can centralize secrets and set up OIDC for secure authentication. This allows you to manage and share sensitive configuration data across multiple stacks efficiently.
+{{% /notes %}}
+
+Click **Save deployment configuration**.
+
+![Animation of how to configure drift detection in Pulumi](./pulumi-configure-drift.mp4)
+
+Once you have configured Pulumi Deployments, open the Actions menu and choose `Detect Drift`, then click `Deploy` to validate your stack, program and deployments are all running before we introduce drift.
+
+![Screenshot of the Pulumi Cloud console to run a Drift Detection](./pulumi-detect-drift-action.png)
+
+Once the drift run is complete, you should see a new Drift check on the stack `Drift` Tab
+
+## Introduce drift
+
+To see Drift Detection in action, you will now introduce drift by manually making changes to your newly created EC2 resource, in this case by adding a new tag.
+
+Log in to the AWS Management console with your browser and ensure you are in the correct AWS region where your Pulumi stack is deployed.
+
+Modify your EC2 Instance by going to the EC2 Dashboard, then select the EC2 instance named `webserver` and click on the **Tags** tab. Next, add a tag by clicking the **Manage tags** button. Then click the **Add new tag** button with the key `Description` and value of `Pulumi Drift Detection`. Finally, **Save** the changes.
+
+## Run a Drift Detection deployment
+
+Return to the Pulumi Cloud and your stack, and again choose the **Detect Drift** action and click **Deploy**.
+
+After your run completes you will see a warning icon on the Drift tab, indicating that drift has been detected. You can see a detailed summary of what resources have been updated or deleted, the properties that have changed, when the drift run happened and a link to the Deployment with further details.
+
+![Animation of the Pulumi Cloud console Drift tab](./pulumi-drift-detected.mp4)
+
+## Remediate Drift
+
+Now that you have detected and reviewed your infrastructure drift, you can choose to remediate and overwrite the changes made in your cloud provider with the most recently specified desired state of your Pulumi program. This can be done via the **Remediate drift** option in the **Actions** menu on your stack page.
+
+Alternatively, you can accept the changes made in your cloud provider and apply them to your Pulumi state. This can be done via the **Refresh** option in the **Actions** drop down on your stack page. You will also want to update your Pulumi program to align with the changes made in the cloud provider.
```

Comment:
Is there no option for having it auto update your Pulumi program for you or generating the right code to replace it?

Reply:
I don't think there is for refresh as far as I know, unless this was something recently added. It only updates the state to true it up with the current state of the deployed resources, but does not give the code to update the resource's configuration in the program.

Reply:
I don't think there is for refresh as far as I know, unless this was something recently added. It only updates the state to true it up with the current state of the deployed resources, but does not give the code to update the resource's configuration in the program.

Reply:
That's correct, it does not give you the right code. Just shows you what's changed and you can either choose to overwrite the changes, or accept them and update the state.